lib/usd/utils/util.cpp

```diff
@@ -101,4 +101,20 @@ getPrimSpecAtEditTarget(UsdStageWeakPtr stage, const UsdPrim& prim)
     return stage->GetEditTarget().GetPrimSpecForScenePath(prim.GetPath());
 }
 
+bool
+isInternalReference(const SdfPrimSpecHandle& primSpec)
+{
+    bool isInternalRef{false};
+
+    for (const SdfReference& ref : primSpec->GetReferenceList().GetAddedOrExplicitItems()) {
+        // GetAssetPath returns the asset path to the root layer of the referenced layer
+        // this will be empty in the case of an internal reference.
+        if (ref.GetAssetPath().empty()) {
+            isInternalRef = true;
```

Comment:
Don't you want to
break;
at this point?

Reply:
@ppt-adsk see 610f2ce099cceab4f2727e7df0774aa39d7ea815